h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoder.java

```diff
@@ -520,27 +565,30 @@ public void map(Chunk cs[]) {
      * @param foldColumnName column's name that contains fold number the row is belong to.
      * @param withBlendedAvg whether to apply blending or not.
      * @param noiseLevel amount of noise to add to the final encodings.
-     * @param imputeNAsWithNewCategory set to `true` to impute NAs with new category.
      * @param seed we might want to specify particular values for reproducibility in tests.
      * @return copy of the `data` frame with encodings
      */
     public Frame applyTargetEncoding(Frame data,
                                      String targetColumnName,
                                      Map<String, Frame> columnToEncodingMap,
-                                     byte dataLeakageHandlingStrategy,
+                                     DataLeakageHandlingStrategy dataLeakageHandlingStrategy,
                                      String foldColumnName,
                                      boolean withBlendedAvg,
                                      double noiseLevel,
-                                     boolean imputeNAsWithNewCategory,
-                                     long seed) {
+                                     long seed,
+                                     Key<Frame> encodedFrameKey) {
```

Comment:
Plase note `imputeNAsWithNewCategory` was unused parameter. I've removed it and with it made the appropriate adjustments to other parts of code (including tests).

My model API call would have to provide the value, so it was interfering with any new code. That's why it is removed in this API.

Reply:
@Pscheidl `imputeNAsWithNewCategory` was hanging there due to a limitation in our merge implementation. Take a look at jira https://0xdata.atlassian.net/browse/PUBDEV-6729. The point is that we in the end of the day want this option/parameter to be available for users.

Reply:
When I think about it, Sparkling Water might integrate on it and this would break the API for them.

Otherwise, the preferred way to access TE from now now would be Model and its parameters. If there was no SW, we could dump the parameter anyway for now and move it back there when it actually does something.

Reply:
I agree, just giving you more context.